Jenkinsfile

```diff
@@ -91,51 +93,67 @@ pipeline {
+                            sh "make test PARALLEL_COUNT=2"
+                        }
                     }
                 }
 
                 stage('Cache') {
                     steps {
-                        sh "make cache-test PARALLEL_COUNT=2"
+                        catchError (buildResult: 'FAILURE', stageResult: 'FAILURE') {
+                            sh "make cache-test PARALLEL_COUNT=2"
+                        }
                     }
                 }
 
                 stage('Batch') {
                     steps {
-                        sh 'make batch-test PARALLEL_COUNT=2'
+                        catchError (buildResult: 'FAILURE', stageResult: 'FAILURE') {
+                            sh 'make batch-test PARALLEL_COUNT=2'
+                        }
                     }
                 }
 
                 stage('Multinamespace') {
                     steps {
-                        sh "kubectl config use-context $TESTING_CONTEXT"
-                        sh 'make multinamespace-test'
+                        catchError (buildResult: 'FAILURE', stageResult: 'FAILURE') {
+                            sh "kubectl config use-context $TESTING_CONTEXT"
+                            sh 'make multinamespace-test'
+                        }
                     }
                 }
 
                 stage('Backup/Restore') {
                     steps {
-                        sh 'make backuprestore-test INFINISPAN_CPU=500m'
+                        catchError (buildResult: 'FAILURE', stageResult: 'FAILURE') {
+                            sh 'make backuprestore-test INFINISPAN_CPU=500m'
+                        }
                     }
                 }
 
                 stage('Hot Rod Rolling Upgrade') {
                     steps {
-                         sh 'make hotrod-upgrade-test'
+                        catchError (buildResult: 'FAILURE', stageResult: 'FAILURE') {
+                            sh 'make hotrod-upgrade-test'
+                        }
                     }
                 }
 
                 stage('Upgrade') {
                     steps {
-                        sh 'make upgrade-test SUBSCRIPTION_STARTING_CSV=infinispan-operator.v2.2.1'
+                        catchError (buildResult: 'FAILURE', stageResult: 'FAILURE') {
+                            sh 'make upgrade-test SUBSCRIPTION_STARTING_CSV=infinispan-operator.v2.2.1'
+                        }
                     }
                 }
 
                 stage('Xsite') {
                     steps {
-                        sh 'scripts/ci/configure-xsite.sh'
-                        sh 'INFINISPAN_MEMORY="1Gi" go test -v ./test/e2e/xsite/ -timeout 45m'
+                        catchError (buildResult: 'FAILURE', stageResult: 'FAILURE') {
+                            sh 'scripts/ci/configure-xsite.sh'
+                            sh 'INFINISPAN_MEMORY="1Gi" ./scripts/run-tests.sh xsite 45m'
```

Comment:
I think `GO_JUNIT_REPORT` is not visible outside the `Makefile` commands and that is causing the "issues" in Jenkins. Maybe try to make a new target to run the xsite test suite.